### PR TITLE
Improve validation errors for assay attachments

### PR DESCRIPTION
**Pull request by straight-shoota**
This resolves part of #1456. Nice error messages related to assay files are not yet resolved, pending a revisit of the feature.

### CODE REVIEW

**Review by ysbaddaden — APPROVED** [review]
:+1:

**Comment by diegoliberman** [comment]
@straight-shoota I made a mistake.  The LOINC code isn't mandatory.  I don't know what I was thinking.  Thanks @devduarte for spotting this.

**Review by devduarte — CHANGES_REQUESTED** [review]

File: app/models/assay_attachment.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ class AssayAttachment < ActiveRecord::Base
   after_save :assign_assay_file
 
   validate :presence_assay_file_or_result
+  validates_presence_of :loinc_code
```
[the remark this answers is not in the document]
Comment (diegoliberman) [reply app/models/assay_attachment.rb:10]: DiegoD is right, the LOINC code isn't mandatory.

[the remark this answers is not in the document]
Comment (ysbaddaden) [reply app/models/assay_attachment.rb:10]: @diegoliberman But when present, we should verify that its value is valid?

**Review by devduarte — APPROVED** [review]
🚀 🍟

**Review by diegoliberman — APPROVED** [review]

**Pull request merged**